### ISSUE TITLE
io.qua.boo.res.mav.BootstrapMavenContext: Unrecognised tag: 'blocked'

### ISSUE THREAD

**Issue by samodadela** [quarkusio/quarkus#17150]
## Describe the bug
When starting quarkus:dev:
```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-05-11 20:20:47,313 WARN  [io.qua.boo.res.mav.BootstrapMavenContext] (main) Settings problem encountered at /opt/maven/conf/settings.xml, line 164, column 16: org.apache.maven.settings.io.SettingsParseException: Unrecognised tag: 'blocked' (position: START_TAG seen ...</url>\r\n      <blocked>... @164:16) 
        at org.apache.maven.settings.io.DefaultSettingsReader.read(DefaultSettingsReader.java:87)
        at org.apache.maven.settings.building.DefaultSettingsBuilder.readSettings(DefaultSettingsBuilder.java:182)
        at org.apache.maven.settings.building.DefaultSettingsBuilder.build(DefaultSettingsBuilder.java:104)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getEffectiveSettings(BootstrapMavenContext.java:252)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.isOffline(BootstrapMavenContext.java:218)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.newRepositorySystem(BootstrapMavenContext.java:648)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:223)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:124)
        at io.quarkus.bootstrap.BootstrapAppModelFactory.getAppModelResolver(BootstrapAppModelFactory.java:192)
        at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:305)
        at io.quarkus.bootstrap.app.QuarkusBootstrap.bootstrap(QuarkusBootstrap.java:160)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:143)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: org.codehaus.plexus.util.xml.pull.XmlPullParserException: Unrecognised tag: 'blocked' (position: START_TAG seen ...</url>\r\n      <blocked>... @164:16)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.checkUnknownElement(SettingsXpp3Reader.java:149)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.parseMirror(SettingsXpp3Reader.java:920)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.parseSettings(SettingsXpp3Reader.java:1416)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.read(SettingsXpp3Reader.java:1544)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.read(SettingsXpp3Reader.java:566)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.read(SettingsXpp3Reader.java:595)
        at org.apache.maven.settings.io.DefaultSettingsReader.read(DefaultSettingsReader.java:83)
        ... 12 more
```

### Expected behavior
No warning should be printed.

### Actual behavior
Warning is printed. App starts anyway.

## To Reproduce
```
mvn clean quarkus:dev 
```

### Output of `uname -a` or `ver`
```
Linux a 5.12.1-2-MANJARO #1 SMP PREEMPT Sun May 2 13:56:40 UTC 2021 x86_64 GNU/Linux
```

### Output of `java -version`
```
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)
```

### Quarkus version or git rev
Quarkus 1.11.6.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
mvn -version
Apache Maven 3.8.1 (NON-CANONICAL_2021-04-26T21:52:54Z_root)
Maven home: /opt/maven
Java version: 11, vendor: Oracle Corporation, runtime: /opt/openjdk-11
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.12.1-2-manjaro", arch: "amd64", family: "unix"
```

## Additional context
Nothing... suspecting maven is too new, though.

/opt/maven/conf/settings.xml (the problem is the `<blocked>` tag, below)
```
  <mirrors>
    <!-- mirror
     | Specifies a repository mirror site to use instead of a given repository. The repository that
     | this mirror serves has an ID that matches the mirrorOf element of this mirror. IDs are used
     | for inheritance and direct lookup purposes, and must be unique across the set of mirrors.
     |
    <mirror>
      <id>mirrorId</id>
      <mirrorOf>repositoryId</mirrorOf>
      <name>Human Readable Name for this Mirror.</name>
      <url>http://my.repository.com/repo/path</url>
    </mirror>
     -->
    <mirror>
      <id>maven-default-http-blocker</id>
      <mirrorOf>external:http:*</mirrorOf>
      <name>Pseudo repository to mirror external repositories initially using HTTP.</name>
      <url>http://0.0.0.0/</url>
      <blocked>true</blocked>
    </mirror>
  </mirrors>
```

**Comment by gsmet** [comment]
This is due to using Maven 3.8. We are using some Maven libraries in Quarkus and they are not supporting a new tag introduced in Maven 3.8.

This has been resolved in the upcoming Quarkus 2.0 as we upgraded our internals to 3.8 but for 1.x, the only way to get rid of the warning is to use Maven 3.6.3.

**Comment by gwenneg** [comment]
@gsmet As you may already know, Maven `3.8.1` fixed [CVE-2021-26291](https://maven.apache.org/docs/3.8.1/release-notes.html#cve-2021-26291) with the new `blocked` tag which is causing the warning. I Quarkus still exposed to that CVE (because the tag is not supported) when Maven `3.8.1` is used or not?
